src/org/zaproxy/zap/extension/callback/ui/CallbackTable.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2017 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.callback.ui;
+
+import org.jdesktop.swingx.renderer.DefaultTableRenderer;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.view.ZapTable;
+import org.zaproxy.zap.view.renderer.DateFormatStringValue;
+
+import javax.swing.ListSelectionModel;
+import javax.swing.SortOrder;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+import javax.swing.table.TableModel;
+import java.util.Date;
+
+public class CallbackTable extends ZapTable {
+
+    private static final long serialVersionUID = 1L;
+
+    public CallbackTable(CallbackTableModel model) {
+        super(model);
+
+        createDefaultColumnsFromModel();
+        initializeColumnWidths();
+
+        setSelectionMode(ListSelectionModel.MULTIPLE_INTERVAL_SELECTION);
+
+        setSortOrderCycle(SortOrder.ASCENDING, SortOrder.DESCENDING, SortOrder.UNSORTED);
+
+        setColumnSelectionAllowed(false);
+        setCellSelectionEnabled(false);
+        setRowSelectionAllowed(true);
+
+        setAutoCreateRowSorter(true);
+
+        getSelectionModel().addListSelectionListener(new CallbackTable.DisplayMessageOnSelectionValueChange());
+    }
+
+    @Override
+    protected void createDefaultRenderers() {
+        super.createDefaultRenderers();
+        setDefaultRenderer(Date.class, new DefaultTableRenderer(new DateFormatStringValue()));
+    }
+
+    @Override
+    public void setModel(TableModel tableModel) {
+        if (!(tableModel instanceof CallbackTableModel)) {
+            throw new IllegalArgumentException("Parameter tableModel must be a CallbackTableModel.");
+        }
+
+        super.setModel(tableModel);
+    }
+
+    @Override
+    public CallbackTableModel getModel() {
+        return (CallbackTableModel) super.getModel();
+    }
+
+    private class DisplayMessageOnSelectionValueChange implements ListSelectionListener {
+
+        @Override
+        public void valueChanged(final ListSelectionEvent evt) {
+            if (!evt.getValueIsAdjusting()) {
+                CallbackModel callbackModel = getModel().getCallbackModelAt(getSelectedRow());
```

Comment:
This will lead to an exception (`ArrayIndexOutOfBoundsException: -1`) when there's no selection (for example, clear the table with an entry selected).